documents/ARCHITECTURE.md

```diff
@@ -0,0 +1,80 @@
+# Stickian Empire - The Architecture
+
+This document holds the key architectural components that the developers of the Stickian Empire game follow. The reading is dense, and the writer is denser, but don't worry, nothing goes over his head, his reflexes are too quick.
+
+Take with a hint of salt any point given as a draft with the status :page_facing_up:.
+
+Work in progress... :warning: :construction_worker:
+
+## Index
+
+1. [High-Level Architecture](#high-level-architecture)
+2. [Tools](#tools)
+3. [Back-end Architecture](#back-end-architecture)
+4. [The NoSQL Approach](#the-nosql-approach)
+5. [Real(bothersome)-time problem](#realbothersome-time-problem)
+6. ["Micro" Front-ends](#micro-front-ends)
+7. [Front-end state management](#front-end-state-management)
+
+## High-Level Architecture
+
+To be able to provide fun to thousands, if millions, of (long distant future) players, a robust and future proof architecture is necessary. This first High-Level approach is to guide the developers and contributors to our final goal.
+
+```mermaid
+flowchart LR
+    subgraph frontend
+        home[Login/Home screen]
+        game[Game screen]
+        game[Game screen]
+        game[Game screen]
+        chat[Chat view]
+    end
+    subgraph backend
+        home --- auths
+        auths[Authentication service] --- relationaldb
+        game --- gameb1
+        game --- gameb2
+        game --- gameb3
+        gameb1[Game service X] --- gamedb
+        gameb2[Game service Y] --- gamedb
+        gameb3[Game service Z] --- gamedb
+        chat --- chatb
+        chatb[Chat service] --- relationaldb
+        gamedb[(NoSQL Database)]
+        relationaldb[(Relational Database)]
+    end
+```
+
+## Tools
+
+The tools for this project are stated as follows:
+
+- [Go](https://go.dev/) as one of the simplest back-end language with an enourmous power, but a compiled environment to ensure types and no runtime hiccups (yes, I'm looking at you Python 3.x :eyes:).
+- [React](https://reactjs.org/) which can be considered as good as any popular front-end framework :poop:, it is MANDATORY to use it with [Typescript](https://www.typescriptlang.org/), forget about classes and go for hooks and functional components.
```

Comment:
React is a great option indeed. However, adding mobile/tablet support for the game will give some extra work. I suggest you look up some tools such as [Flutter](https://flutter.dev/), that allow easy development for multi-platform apps.

Reply:
I do know Flutter and work with it daily, just that it can get a bit more messy than React. If we ever want to introduce mobile support, I'd say we do a native implementation, browser experience will never need the same UI components than a mobile experience. One-fits-all is not always the best solution for UX/UI. Mind you, the back-end can be the same, so mobile players can play with browser players in the end.